bindings/src/main/java/io/pravega/storage/azure/AzureBlobClientImpl.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.storage.azure;
+
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.azure.storage.blob.models.BlobErrorCode;
+import com.azure.storage.blob.models.AppendBlobItem;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import com.azure.storage.blob.models.BlobProperties;
+import com.azure.storage.blob.models.BlobRange;
+import com.azure.storage.blob.models.AppendBlobRequestConditions;
+import com.azure.storage.blob.specialized.AppendBlobClient;
+import com.azure.storage.blob.specialized.BlobClientBase;
+import com.azure.storage.common.StorageSharedKeyCredential;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import java.io.InputStream;
+
+@Slf4j
```

Comment:
While looking at your PR I realized that because we abstracted out azure interaction by introducing `AzureClient ` interface and implemented our own mock - a strange thing happened with code coverage ..actual azure code `AzureBlobClientImpl` had 0% coverage - because we never call real azure in unit tests !!!
To make it worse ...The big complication there is that azure SDK classes are final :(
But worry not - I figured out a way to mock final classes
Here is how I'm doing it
branch https://github.com/sachin-j-joshi/pravega/tree/azure_bindings
commit https://github.com/sachin-j-joshi/pravega/commit/2eedef58fc157c09f30794adf87b24b7666beb49
Pease take this code and build on top of it and write tests for rest of the methods in AzureBlobClientImpl to cover it 100%